lib/dry/monitor/notifications.rb

```diff
@@ -37,17 +32,15 @@ def stop(event_id, payload)
         instrument(event_id, payload)
       end
 
-      def instrument(event_id, payload = EMPTY_HASH, &block)
-        if block
-          result, time = clock.measure(&block)
+      def instrument(event_id, payload = EMPTY_HASH)
+        if block_given?
+          result, time = @clock.measure { yield }
         end
 
+        payload[:time] = time if time
```

Comment:
this could be improved and moved to block_given? One less check to perform

Reply:
so, we can't mutate this hash here, payload is typically created outside of the library, and we can't assume it's safe to mutate it here.

Reply:
and btw it has `EMPTY_HASH` as a default which is frozen